portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,57 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+ 
+import com.google.sps.classes.CommentSection;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import com.google.gson.Gson;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+ 
+/** Servlet that returns a comment section. */
+@WebServlet("/comment-section")
 public class DataServlet extends HttpServlet {
+ 
+  private CommentSection commentSection = new CommentSection();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.setContentType("application/json");
+    String json = new Gson().toJson(commentSection);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the input from the form and log it.
+    String comment = getComment(request);
+    commentSection.logComment(comment);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+  
+  /** Returns the comment entered by the user */
+  private String getComment(HttpServletRequest request) {
+    // Get the input from the form.
+    String commentString = request.getParameter("comment");
+
+    return commentString;
```

Comment:
you can return `request.getParameter("comment");` directly

Reply:
done! I am using getParameter() directly